compiler/rustc_resolve/src/imports.rs

```diff
@@ -348,6 +347,21 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                 resolution.binding = Some(binding);
             }
 
+            if let Some(old_binding) = resolution.shadowed_glob && binding.is_glob_import() {
+                assert!(old_binding.is_glob_import());
+                if res != old_binding.res() && key.ns != TypeNS {
```

Comment:
Removing ﻿`key.nx != TypeNS` will cause #109153 to throw an ambiguity error. I am not sure whether it is better to let the check pass or to throw an error.

Reply:
>Removing ﻿`key.nx != TypeNS` will cause #109153 to throw an ambiguity error.

That seems entirely fine to me.

Reply:
Although I don't understand why `shadowed_glob` assignment was moved from the place it previously was to here.

Reply:
> Although I don't understand why shadowed_glob assignment was moved from the place it previously was to here.

The issue seems to be simply a difference in code style? There should be no difference in logic between the two

Reply:
It does lead to useless code execution on both sides and I will put it back